lib/recaptcha.ex

```diff
@@ -1,75 +1,48 @@
 defmodule Recaptcha do
-  require Elixir.EEx
+  @moduledoc """
+    A module for verifying reCAPTCHA version 2.0 response strings.
 
-  @secret_key_errors ~w(missing-input-secret invalid-input-secret)
-
-  EEx.function_from_file :defp, :render_template, "lib/template.html.eex", [:assigns]
-
-  @doc """
-  Returns a string with reCAPTCHA code
-
-  To convert the string to html code, use Phoenix.HTML.Raw/1 method
+    See the [documentation](https://developers.google.com/recaptcha/docs/verify) for more details.
   """
-  def display(options \\ []) do
-    public_key = options[:public_key] || config.public_key
-    render_template(public_key: public_key, options: options)
-  end
+  @secret Application.get_env(:recaptcha, :secret)
+  @http_client Application.get_env(:recaptcha, :http_client, Recaptcha.Http)
 
   @doc """
-  Verifies reCAPTCHA response string.
+  Verifies a reCAPTCHA response string.
 
-  The function returns :ok or :error, depending on the verification's result
+  ## Options
 
-  :ok is returned when the response code was successfully verified
+    * `:timeout` - the timeout for the request (defaults to 5000ms)
+    * `:secret`  - the secret key used by recaptcha (defaults to the secret provided in application config)
+    * `:remote_ip` - the IP address of the user (optional and not set by default)
 
-  :error is returned when the response is nil or if the reCAPTCHA server returned
-  a missing-input-response or invalid-input-response code
+  ## Example
 
-  The function raises RuntimeError if the server returned a missing-input-secret or invalid-input-secret
-  code
+    {:ok, api_response} = Recaptcha.verify("response_string")
   """
-  def verify(remote_ip, response, options \\ [])
+  @spec verify(String.t, [timeout: integer, secret: String.t, remote_ip: String.t]) :: {:ok, map} | {:error, [String.t]}
+  def verify(response, options \\ [])
 
-  def verify(remote_ip, response, options) when is_tuple(remote_ip) do
-    verify(:inet_parse.ntoa(remote_ip), response, options)
+  def verify(nil = _response, _) do
+    {:error, ["invalid-input-response"]}
   end
 
-  def verify(_remote_ip, nil = _response, _options), do: :error
-
-  def verify(remote_ip, response, options) do
-    case api_response(remote_ip, response, options)  do
-      %{"success" => true} ->
-        :ok
-      %{"success" => false, "error-codes" => error_codes} ->
-        handle_error_codes(error_codes)
-      %{"success" => false} ->
-        :error
+  def verify(response, options) do
+    case @http_client.request_verification(request_body(response, options), Keyword.take(options, [:timeout])) do
```

Comment:
ok yep thats fair enough, so we'll handle errors from HttpPoison and Poison and hand them on in the `errors` then